Tests/Integration/DevAddrCacheTest.cs

```diff
@@ -237,12 +237,13 @@ public async Task Multi_Gateway_When_DevAddr_Is_Not_In_Cache_Query_Iot_Hub_Only_
 
             var deviceGetter = new DeviceGetter(registryManagerMock.Object, this.cache);
             // Simulate three queries
+            var devNonce = new DevNonce(0xABCD);
             var tasks = new Task[4]
               {
-                deviceGetter.GetDeviceList(null, gateway1, "ABCD", devAddrJoining),
-                deviceGetter.GetDeviceList(null, gateway1, "ABCD", devAddrJoining),
-                deviceGetter.GetDeviceList(null, gateway2, "ABCD", devAddrJoining),
-                deviceGetter.GetDeviceList(null, gateway2, "ABCD", devAddrJoining)
+                deviceGetter.GetDeviceList(null, gateway1, devNonce, devAddrJoining),
```

Comment:
nit: Perhaps we can use this opportunity to simplify the whole query instead of extracting the repeated device nonce into a variable:

```c#
var tasks =
    from gw in new[] { gateway1, gateway2 }
    select Enumerable.Repeat(gw, 2) into gws // repeat each gateway twice
    from gw in gws
    select deviceGetter.GetDeviceList(null, gw, new DevNonce(0xABCD), devAddrJoining);
```

What do you think?

Reply:
Done in e5b0eb53bd64a93a01c97f8a125a504d89ea93c8